template.yaml

```diff
@@ -125,6 +126,7 @@ Resources:
         HTTPRequestRoots:
           Type: Api
           Properties:
+            RestApiId: !Ref FrontendAPI
```

Comment:
ECApiFunction has two events, but I've only added `RestApiId: !Ref FrontendAPI` to one of them. This will break the frontend site until the 2nd PR is merged, but I can't figure out a different workaround.

The reason that I do this is because if I added it to both events, then the implicit API gateway would no longer be automatically created. And because that implicit API gateway is referenced in the output, not creating it will cause the deploy to fail. 

And to further complicate things, in order to create the new `FrontendAPI` gateway, I need to give it at least one method, otherwise it will fail to create. 

So I've opted for the above: only changing one Event to reference `FrontendAPI` so that the implicit gateway is still being created and `FrontendAPI` has at least one method so it will successfully create. And I clean up with the 2nd deploy by changing the 2nd event to reference `FrontendAPI`.